gr-fec/lib/ldpc_G_matrix_impl.cc

```diff
@@ -24,21 +22,23 @@ ldpc_G_matrix::sptr ldpc_G_matrix::make(const std::string filename)
     return ldpc_G_matrix::sptr(new ldpc_G_matrix_impl(filename));
 }
 
-ldpc_G_matrix_impl::ldpc_G_matrix_impl(const std::string filename) : fec_mtrx_impl()
+ldpc_G_matrix_impl::ldpc_G_matrix_impl(const std::string filename)
+    : fec_mtrx_impl(), d_logger(fmt::format("ldpc_G_matrix {}", filename))
 {
-    configure_default_loggers(d_logger, d_debug_logger, "ldpc_G_matrix");
-
     // Read the matrix from a file in alist format
     matrix_sptr x = read_matrix_from_file(filename);
     d_num_cols = x->size2;
     d_num_rows = x->size1;
 
+    // FIXME: totally misled. If we can't read x safely here, then we also can't copy it
+    // savely. Don't make that copy. All we do is checking the first d_n×d_n submatrix for
+    // identity-matrixness, then transposing and storing the result.
+    //
```

Comment:
I don't understand this comment accurately, even after having read through the code below. Chances are that the next person that stumbles upon this comment, maybe years from now, will find it a bit cryptic too. Can we try to reword?

What is the problem that the comment tries to describe? Is it that `read_matrix_from_file()` might fail, in which case the function should exit early without trying to do anything? I agree that the check for whether G is of the form [I P] is bad code, but there is a FIXME comment specific to that bad code, and the bad code is technically correct (in the sense that it doesn't have bugs and produces correct results).

Reply:
what is done here is copying the input matrix into a temporary, specifically allocated matrix within this function, "because we can't know what someone else might be doing with the pointer-passed matrix". 

Well, if we can't know that the matrix is not modified while this function works on it, we have a severe multithreading bug. Copying the matrix at the beginning reduces the time window when that might manifest as data corruption, but it's not an appropriate mitigation.

Reply:
I see where you're going, but I still don't understand this (either your comment or the comment in the original code that says `Make an actual copy so we guarantee that we're not sharing memory with another class that reads the same alist file.`). Assuming that the file itself isn't being altered while it is being read, and unless I've missed something. I don't see any potential problems and no need to copy the matrix. `read_matrix_from_file()` allocates a new matrix and then passes ownership of that to the caller by returning a sptr. There is apparent problem if there are concurrent calls to `read_matrix_from_file()` on different threads, whether they read the same alist file or a different one (why would it matter if the alist file is the same? it is just a path that gets read).

Reply:
exactly! the "Make an actual copy…" is from the original code. Will rephrase that to be more explictly a `FIXME: remove copy". My interpretation of what the original author got wrong about ownership doesn't really matter.